### PR TITLE
Terraform v0.13

### PR DESCRIPTION
**Pull request by bendrucker**
This pull request updates TFLint for Terraform v0.13 compatibility and will serve as a tracking branch during the [0.13 beta](https://github.com/hashicorp/terraform/issues/25016). 

**Status:** Compatible with Terraform 0.12 configurations, and should run successfully with new 0.13 features (`module.{for_each,count}`). Attributes that use `each` or `count` will be unknown and will not be evaluated, even if the `for_each` or `count` expression is static.
 
<hr>

TFLint uses Terraform as a library and never calls the `terraform` binary. Because there are no breaking [changes](https://github.com/hashicorp/terraform/blob/master/CHANGELOG.md) in the language itself (only the CLI), TFLint should remain compatible with Terraform modules written for Terraform 0.12.x. 

## Building

Currently, you must build TFLint locally to use it with Terraform 0.13 modules. We may consider publishing release candidates when Terraform does the same.

```sh
make build
```

## Issues

For specific issues (crashes, errors, etc.), please open a new issue and reference that you are testing with 0.13. Please include example configuration.

For general discussion, please comment on the main issue (#770).

## Pull Requests

If you'd like to help prepare TFLint for Terraform 0.13, please target PRs against the `terraform-v0.13` branch (which this PR tracks).

### CODE REVIEW

**Comment by bendrucker** [comment]
Left a TODO behind about next steps for getting this working. TFLint currently can't find any modules as a consequence of `for_each` changing around the structure.

**Comment by bendrucker** [comment]
This is now in a working state. Somewhat predictably, the `each` and `count` objects are not defined in the parent context, so a module variable that uses either will be unknown. Evaluating `for_each`/`count` and expanding modules to module instances will take a more ambitious refactoring effort. I've added a test showing how using `for_each` prevents an invalid instance type from being detected. With more robust support for module iteration, that configuration should result in an issue referencing the full path, including the `each.key` value.

**Comment by bendrucker** [comment]
Terraform 0.13 is now GA, I'll get this ready for release

File: tflint/runner.go [hunk 1]
```diff
@@ -51,22 +51,25 @@ func NewRunner(c *Config, files map[string]*hcl.File, ants map[string]Annotation
...
-					Env: getTFWorkspace(),
-				},
-				Config:             cfg.Root,
-				VariableValues:     prepareVariableValues(cfg, variables...),
-				VariableValuesLock: &sync.Mutex{},
-			},
-		},
+		// TODO: As described in the godoc for UnkeyedInstanceShim,
+		// it will need to be replaced now that module.for_each is supported
+		ctx:         ctx.WithPath(cfg.Path.UnkeyedInstanceShim()),
```
Comment (wata727) [inline tflint/runner.go:72]: 👍

**Pull request merged**